src/coordinated_workers/coordinator.py

```diff
@@ -469,19 +488,154 @@ def _reconcile(self):
+            return worker_telemetry.proxy_tracing_receivers_urls(
+                hostname=self.app_hostname(
+                    self.hostname, self._charm.app.name, self._charm.model.name
+                ),
+                proxy_worker_telemetry_port=proxy_worker_telemetry_port,
+                tls_available=self.tls_available,
+                tracing_target_type=_type,
+                protocols=[receiver.protocol.name for receiver in receivers],  # type: ignore
+            )
+
+        return {receiver.protocol.name: receiver.url for receiver in receivers}  # type: ignore
+
     @property
     def _charm_tracing_receivers_urls(self) -> Dict[str, str]:
-        """Returns the charm tracing enabled receivers with their corresponding endpoints."""
-        endpoints = self.charm_tracing.get_all_endpoints()
-        receivers = endpoints.receivers if endpoints else ()
-        return {receiver.protocol.name: receiver.url for receiver in receivers}
+        """Returns the charm tracing enabled receivers per tracing protocol with their corresponding endpoints to be published into the cluster data.
+
+        When worker telemetry proxy is enabled and when not self tracing, returns the proxy url for sending traces.
+
+        Returns:
+            A dictionary of tracing protocols and the respective tracing receiver urls.
+            {
+                "otlp_http": "http://tempo.tempo.svc.cluster.local:4318",
+            }
+
+            Or when worker telemetry proxy is enabled,
+            {
+                "otlp_http": "http://tempo-0.tempo-endpoints.tempo.svc.cluster.local:3300/proxy/charm-tracing/otlp_http/",
+            }
+        """
+        return self._tracing_receivers_urls(self.charm_tracing, "charm-tracing")
 
     @property
     def _workload_tracing_receivers_urls(self) -> Dict[str, str]:
-        """Returns the workload tracing enabled receivers with their corresponding endpoints."""
-        endpoints = self.workload_tracing.get_all_endpoints()
-        receivers = endpoints.receivers if endpoints else ()
-        return {receiver.protocol.name: receiver.url for receiver in receivers}
+        """Returns the workload tracing enabled receivers with their corresponding endpoints to be published into the cluster data.
+
+        When worker telemetry proxy is enabled and when not self tracing, returns the proxy url for sending traces.
+
+        Returns:
+            A dictionary of tracing protocols and the respective tracing receiver urls.
+            {
+                "otlp_http": "http://tempo.tempo.svc.cluster.local:4318",
+            }
+
+            Or when worker telemetry proxy is enabled,
+            {
+                "otlp_http": "http://tempo-0.tempo-endpoints.tempo.svc.cluster.local:3300/proxy/workload-tracing/otlp_http/",
+            }
+        """
+        return self._tracing_receivers_urls(self.workload_tracing, "workload-tracing")
+
+    @property
+    def remote_write_endpoints(self):
```

Comment:
```suggestion
    def remote_write_endpoints(self) -> Optional[List[str]]:
```

didn't the type checker catch this? Can you check it's running?